routes/order.js

```diff
@@ -0,0 +1,56 @@
+const express = require('express');
+const router = express.Router();
+const Order = require('../models/orderModel');
+const mongoose = require('mongoose');
+
+router.post('/', async (req, res) => {
+  console.log(req)
+  const {
+    fname,
+    lname,
+    street_address,
+    city,
+    postal_code,
+    items,
+    deliveryDate,
+  } = req.body.payload;
+
+  const randomInt = (min, max) => {
+    return Math.floor(Math.random() * (max - min));
+  };
+
+  db = mongoose.connection.db;
+
+  const checkOrderNumber = async (number) => {
+    let result = await db.collection('orders').findOne({ orderNumber: number });
+    return result;
+  };
+  let orderNumber;
+  let repeat;
+  do {
+    orderNumber = randomInt(100000, 999999);
+    repeat = checkOrderNumber(orderNumber) ? false : true;
+  } while (repeat);
+
+  let order = new Order({
+    fname: fname,
+    lname: lname,
+    items: items,
+    orderNumber: orderNumber,
```

Comment:
you can use the order records' id which is guaranteed to be unique instead of generating an order number yourself

Reply:
Only reason I'm doing it this way is because I want to display an order number eventually to the user and the _id is very long string containing both letters and numbers. Thought it would look nicer if it was just a 6 digit number